setup.py

```diff
@@ -87,15 +61,24 @@ def cuda_extension(build_all=False):
             nvcc_args.append("-gencode=arch=compute_80,code=sm_80")
         if cuda_version >= 11.1:
             nvcc_args.append("-gencode=arch=compute_86,code=sm_86")
+        if cuda_version >= 11.8:
+            nvcc_args.append("-gencode=arch=compute_89,code=sm_89")
+        if cuda_version >= 12.0:
+            nvcc_args.append("-gencode=arch=compute_90,code=sm_90")
+
     print("nvcc_args: ", nvcc_args)
     print('-' * 75)
-    include_dirs = [*maybe_download_cub(), os.path.abspath("torchani/cuaev/")]
+    include_dirs = [os.path.abspath("torchani/cuaev/")]
+    # Update C++ standard based on PyTorch version
+    pytorch_version = version.parse(torch.__version__)
+    cxx_args = ['-std=c++17'] if pytorch_version >= version.parse("2.1.0") else ['-std=c++14']
```

Comment:
Updated to make it more robust to compare version. Using string to compare may not give correct result in rare cases
For example
```python
>>> version.parse("2.10.0") >= version.parse("2.2.0")
True
>>> "2.10.0" >= "2.2.0"
False
```